### PR TITLE
OSDOCS#6319: ARM support for GCP

### PR DESCRIPTION
**Pull request by kelbrown20**
**For Version:** 4.14+
**Issue:** [OSDOCS-6319](https://issues.redhat.com//browse/OSDOCS-6319)

**Previews:**  

[Selecting a cluster installation method and preparing it for users -> Supported installation methods for different platforms ](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing-preparing.html)

Installing 
- [Installing a cluster on GCP with customizations -> Tested instance types for GCP on 64-bit ARM infrastructures](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-customizations.html)
- [Installing a cluster on GCP with Network customizations -> Tested instance types for GCP on 64-bit ARM infrastructures](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-network-customizations.html#installation-gcp-tested-machine-types-arm_installing-gcp-network-customizations)
- [Installing a cluster on GCP with Network customizations -> Enabling Shielded VMs](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-network-customizations#installation-gcp-enabling-shielded-vms_installing-gcp-network-customizations)
- [Installing a cluster on GCP in a restricted network -> Tested instance types for GCP on 64-bit ARM infrastructures ](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-restricted-networks-gcp-installer-provisioned.html#installation-gcp-tested-machine-types-arm_installing-restricted-networks-gcp-installer-provisioned)
- [Installing a cluster on GCP into an existing VPC -> Tested instance types for GCP on 64-bit ARM infrastructures ](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-vpc.html#installation-gcp-tested-machine-types-arm_installing-gcp-vpc)
- [Installing a private cluster on GCP -> Tested instance types for GCP in 64-bit ARM infrastructures ](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-private.html#installation-gcp-tested-machine-types-arm_installing-gcp-private)

[Installing on GCP -> Installation configuration parameters for GCP](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installation-config-parameters-gcp#installation-configuration-parameters-optional_installation-config-parameters-gcp)
Updates made in these parameter sections in _Optional Parameters_

-  _Optional Parameters_
            - `compute.architecture`
            - `controlPlane.architecture`
-  _Additional Google Cloud Platform (GCP) configuration parameters_
            - `platform.gcp.defaultMachinePlatform.zones`
            - `controlPlane.platform.gcp.zones`
            - `compute.platform.gcp.zones`

 
**QE review:** 
- [x] QE approval

### CODE REVIEW

**Comment by r4f4** [comment]
@kelbrown20 when I click on `Machine series for 64-bit ARM machines` it just shows me an empty list. Is that expected?

**Comment by r4f4** [comment]
Oh, it's because the Installer PR hasn't merged yet, right? Apart from that
/lgtm

**Comment by kelbrown20** [comment]
@aleskandro @lwan-wanglin Hello! Can you PTAL at this PR from the GCP and ARM QE side?

**Comment by kelbrown20** [comment]
@jeffdyoung Hey Jeff! can you also PTAL at this PR when you get the chance?

File: installing/installing-preparing.adoc [hunk 1]
```diff
@@ -447,7 +452,7 @@ endif::openshift-origin[]
 //This table is for all flavors of OpenShift, except OKD. A separate table is required because OKD does not support multiple AWS architecture types. Trying to maintain one table using conditions, while convenient, is very fragile and prone to publishing errors.
 ifndef::openshift-origin[]
 |===
-||Alibaba |AWS (x86_64) |AWS (arm64) |Azure (x86_64) |Azure (arm64) |Azure Stack Hub |GCP |Nutanix |{rh-openstack} |Bare metal (x86_64) |Bare metal (arm64) |vSphere |VMC |IBM Cloud VPC |{ibmzProductName} |{ibmzProductName} with {op-system-base} KVM |{ibmpowerProductName} |Platform agnostic
+||Alibaba |AWS (64-bit x86) |AWS (64-bit ARM) |Azure (64-bit x86) |Azure (64-bit ARM) |Azure Stack Hub |GCP |Nutanix |{rh-openstack} |Bare metal (64-bit x86) |Bare metal (64-bit ARM) |vSphere |IBM Cloud VPC |{ibmzProductName} |{ibmzProductName} with {op-system-base} KVM |{ibmpowerProductName} |Platform agnostic
```
Comment (lwan-wanglin) [inline installing/installing-preparing.adoc:455]: @kelbrown20 we also support arm upi install on GCP, so we can updage `GCP` to `GCP (64-bit x86) |GCP (64-bit ARM)`

File: modules/installation-configuration-parameters.adoc [hunk 2]
```diff
@@ -639,11 +639,11 @@ ifndef::aws,bare,ibm-z,ibm-power,azure,ibm-power-vs[]
 |String
 endif::aws,bare,ibm-z,ibm-power,azure,ibm-power-vs[]
 
-ifdef::aws,bare,azure[]
+ifdef::aws,bare,azure,gcp[]
 |`controlPlane.architecture`
```
Comment (lwan-wanglin) [inline modules/installation-configuration-parameters.adoc:643]: it seems double `controlPlane.architecture` field in the doc [installation-config-parameters-gcp](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installation-config-parameters-gcp#installation-configuration-parameters-optional_installation-config-parameters-gcp)

File: modules/installation-configuration-parameters.adoc [hunk 3]
```diff
@@ -575,11 +575,11 @@ ifndef::aws,bare,ibm-power,ibm-z,azure,ibm-power-vs[]
 |String
 endif::aws,bare,ibm-power,ibm-z,azure,ibm-power-vs[]
 
-ifdef::aws,bare,azure[]
+ifdef::aws,bare,azure,gcp[]
 |`compute.architecture`
```
Comment (lwan-wanglin) [inline modules/installation-configuration-parameters.adoc:579]: it seems double `compute.architecture` field in the doc [installation-config-parameters-gcp](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installation-config-parameters-gcp#installation-configuration-parameters-optional_installation-config-parameters-gcp)

**Comment by lwan-wanglin** [comment]
@kelbrown20 we now have a bug [OCPBUGS-17904](https://issues.redhat.com/browse/OCPBUGS-17904) for  [enabling shielded VMs](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-network-customizations.html#installation-gcp-enabling-shielded-vms_installing-gcp-network-customizations) arm on gcp, it's a kernel bug, but it influence we install a shielded arm cluster on GCP, should we document it somewhere?

**Comment by kelbrown20** [comment]
Thank you for the review! I just fixed and updated the comments you mentioned 
Regarding: 
> enabling shielded VMs

Is there a different process to enable shielded VMs on ARM? Or is it completely unsupported on ARM?

**Comment by lwan-wanglin** [comment]
> 
> Is there a different process to enable shielded VMs on ARM? Or is it completely unsupported on ARM?

it's completely unsupported on ARM

File: installing/installing-preparing.adoc [hunk 4]
```diff
@@ -497,6 +504,7 @@ ifndef::openshift-origin[]
 |
 |
 |xref:../installing/installing_gcp/installing-restricted-networks-gcp.adoc#installing-restricted-networks-gcp[&#10003;]
+|xref:../installing/installing_gcp/installing-restricted-networks-gcp.adoc#installing-restricted-networks-gcp[&#10003;]
```
Comment (lwan-wanglin) [inline installing/installing-preparing.adoc:507]: we support below `Shared VPC hosted outside of cluster project` on ARM too

**Comment by lwan-wanglin** [comment]
others look good to me! thanks @kelbrown20

File: modules/installation-gcp-enabling-shielded-vms.adoc [hunk 5]
```diff
@@ -14,6 +14,11 @@
 = Enabling Shielded VMs
 You can use Shielded VMs when installing your cluster. Shielded VMs have extra security features including secure boot, firmware and integrity monitoring, and rootkit detection. For more information, see Google's documentation on link:https://cloud.google.com/shielded-vm[Shielded VMs].
 
+[NOTE]
```
Comment (kelbrown20) [inline modules/installation-gcp-enabling-shielded-vms.adoc:17]: @lwan-wanglin I just added this note for the Shielded VM section, WDYT?

File: modules/installation-configuration-parameters.adoc [hunk 6]
```diff
@@ -577,17 +577,17 @@ accounts for the dramatically decreased machine performance.
...
 |`controlPlane.architecture`
 |Determines the instruction set architecture of the machines in the pool. Currently, clusters with varied architectures are not supported. All pools must specify the same architecture. Valid values are `amd64` (the default).
 |String
-endif::aws,bare,ibm-z,ibm-power,azure,ibm-power-vs[]
+endif::aws,bare,gcp,ibm-z,ibm-power,azure,ibm-power-vs[]
 
-ifdef::aws,bare,azure[]
+ifdef::aws,bare,azure,gcp[]
 |`controlPlane.architecture`
 |Determines the instruction set architecture of the machines in the pool. Currently, clusters with varied architectures are not supported. All pools must specify the same architecture. Valid values are `amd64` and `arm64`. See _Supported installation methods for different platforms_ in _Installing_ documentation for information about instance availability.
```
Comment (ShaunaDiaz) [inline modules/installation-configuration-parameters.adoc:588]: Out of scope, really, but maybe you could replace the sentence with an xref to https://docs.openshift.com/container-platform/4.13/installing/installing-preparing.html#supported-installation-methods-for-different-platforms?

**Comment by ShaunaDiaz** [comment]
One suggestion, looks great to me!

File: modules/installation-configuration-parameters.adoc (hunk 6 above)
Comment (kelbrown20) [reply modules/installation-configuration-parameters.adoc:588]: I would want to but Ik we can't pus xrefs in modules :'(

**Pull request merged**